File: Semestrul 4/SDI/lab11/java/TriathlonNetworking/src/main/java/triathlon/network/jsonprotocol/JsonProtocolUtils.java
package triathlon.network.jsonprotocol;

import triathlon.model.*;

import java.util.List;

public class JsonProtocolUtils {
    public static Response createOkResponse() {
        Response resp = new Response();
        resp.setType(ResponseType.OK);
        return resp;
    }

    public static Response createErrorResponse(String errorMessage) {
        Response resp = new Response();
        resp.setType(ResponseType.ERROR);
        resp.setErrorMessage(errorMessage);
        return resp;
    }

    public static Response createResultUpdatedResponse(Result result) {
        Response resp = new Response();
        resp.setType(ResponseType.RESULT_UPDATED);
        resp.setResults(List.of(result));
        return resp;
    }

    public static Response createLoginResponse(RefereeDiscipline asssignement) {
        Response resp = new Response();
        resp.setType(ResponseType.OK);
        resp.setRefereeDiscipline(asssignement);
        return resp;
    }

    public static Response createGetResultsByEventResponse(List<Result> results) {
        Response resp = new Response();
        resp.setType(ResponseType.OK);
        resp.setResults(results);
        return resp;
    }

    public static Request createLoginRequest(Referee referee) {
        Request req = new Request();
        req.setType(RequestType.LOGIN);
        req.setData(referee);
        return req;
    }

    public static Request createLogoutRequest(Referee referee) {
        Request req = new Request();
        req.setType(RequestType.LOGOUT);
        req.setData(referee);
        return req;
    }

    public static Request createGetResultsByEventRequest(Long eventId) {
        Request req = new Request();
        req.setType(RequestType.GET_RESULTS_BY_EVENT);
        req.setData(eventId);
        return req;
    }

    public static Request createUpdateResultRequest(Result result) {
        Request req = new Request();
        req.setType(RequestType.UPDATE_RESULT);
        req.setData(result);
        return req;
    }
    public static Request createGetTotalPointsForParticipantRequest(Long participantId, Long eventId) {
        Request req = new Request();
        req.setType(RequestType.GET_TOTAL_POINTS_FOR_PARTICIPANT);
        req.setData(new Object[]{participantId, eventId});
        return req;
    }

}

File: Semestrul 4/MPP/lab08/TriathlonNetworking/jsonprotocol/JsonProtocolUtils.cs
using System.Collections.Generic;
using Triathlon.Model;

namespace Triathlon.Network.JsonProtocol;

public static class JsonProtocolUtils
{
    public static Response CreateOkResponse()
    {
        Response resp = new Response();
        resp.Type = ResponseType.OK;
        return resp;
    }

    public static Response CreateErrorResponse(string errorMessage)
    {
        Response resp = new Response();
        resp.Type = ResponseType.ERROR;
        resp.ErrorMessage = errorMessage;
        return resp;
    }

    public static Response CreateResultUpdatedResponse(Result result)
    {
        Response resp = new Response();
        resp.Type = ResponseType.RESULT_UPDATED;
        resp.Results = new List<Result> { result };
        return resp;
    }

    public static Response CreateLoginResponse(RefereeDiscipline assignment)
    {
        Response resp = new Response();
        resp.Type = ResponseType.OK;
        resp.RefereeDiscipline = assignment;
        return resp;
    }

    public static Response CreateGetResultsByEventResponse(List<Result> results)
    {
        Response resp = new Response();
        resp.Type = ResponseType.OK;
        resp.Results = results;
        return resp;
    }

    public static Request CreateLoginRequest(Referee referee)
    {
        Request req = new Request();
        req.Type = RequestType.LOGIN;
        req.Data = referee;
        return req;
    }

    public static Request CreateLogoutRequest(Referee referee)
    {
        Request req = new Request();
        req.Type = RequestType.LOGOUT;
        req.Data = referee;
        return req;
    }

    public static Request CreateGetResultsByEventRequest(long eventId)
    {
        Request req = new Request();
        req.Type = RequestType.GET_RESULTS_BY_EVENT;
        req.Data = eventId;
        return req;
    }

    public static Request CreateUpdateResultRequest(Result result)
    {
        Request req = new Request();
        req.Type = RequestType.UPDATE_RESULT;
        req.Data = result;
        return req;
    }

    public static Request CreateGetTotalPointsForParticipantRequest(long participantId, long eventId)
    {
        Request req = new Request();
        req.Type = RequestType.GET_TOTAL_POINTS_FOR_PARTICIPANT;
        req.Data = new object[] { participantId, eventId };
        return req;
    }
}